src/Npgsql/NpgsqlReadBuffer.Stream.cs

```diff
@@ -158,12 +158,50 @@ async Task<int> ReadLong(ValueTask<int> task, bool async)
                 return read;
             }
 
+#if !NET461 && !NETSTANDARD2_0
+            public override int Read(Span<byte> span)
+                => Read(span.ToArray(), CancellationToken.None, false).GetAwaiter().GetResult();
```

Comment:
It won't work in case when the `span` doesn't represent an array from the start to the end. It will allocate a new array and read data into it. Take this method is implemented for streams in .NET Core.

Reply:
Agree, we definitely should not be calling ToArray on any Span/Memory provided by the user - the idea here is to plumb it all the way down.

Reply:
Plumbed :)